src/commands/utils.rs

```diff
@@ -157,23 +191,43 @@ pub fn select_coins_for_spend(
         feerate: FeeRate::from_sat_per_vb(feerate_vb),
         min_fee,
     };
-    if let Err(e) = selector.run_bnb(
-        LowestFee {
-            target,
-            long_term_feerate,
-            change_policy: &change_policy,
-        },
-        100_000,
-    ) {
+    let lowest_fee = LowestFee {
+        target,
+        long_term_feerate,
+        change_policy: &change_policy,
+    };
+    let must_have_change = base_tx.output.is_empty(); // must have change if it is a self-send
```

Comment:
The doc-comment of the function should be updated to document this.

Reply:
I think even better would be to pass it as a parameter, so this function does not have any notion of self-send: it just exposes an interface to select coins such as there is always a change. And we set this in `create_spend` where we already have this notion of self-spend.

Reply:
The new parameter sounds good, will add that.